src/source.ts

```diff
@@ -32,13 +33,30 @@ export default class Source extends React.Component<Props, void> {
 
     if (!map.getSource(this.id) && (geoJsonSource || tileJsonSource)) {
       map.addSource(this.id, (geoJsonSource || tileJsonSource) as any);
+      map.on('load', this.onData);
```

Comment:
I also tried to write a test, but I'm having an hard time with jest 😭 

```tsx
describe('Source', () => {
  let SourceWithContext: any;
  let addLayerMock = jest.fn();
  let addSourceMock = jest.fn();
  let setDataMock = jest.fn();
  let children: any[];
  let childrenWithOneFeature: any[];
  let feature: Object;

  beforeEach(() => {
    addLayerMock = jest.fn();
    addSourceMock = jest.fn();
    setDataMock = jest.fn();
    feature = { coordinates: [-123, 45] };
    children = [{ props: {}}];
    childrenWithOneFeature = [{ props: feature }];

    SourceWithContext = withContext({
      map: React.PropTypes.object
    }, () => ({
      map: {
        addSource: addSourceMock,
        addLayer: addLayerMock,
        on: jest.fn(),
        getSource: jest.fn().mockReturnValue({ setData: setDataMock })
      }
    }))(Source);
  });

  it('Should render source with default options', () => {
    mount(
      <SourceWithContext
        children={children}
      /> as React.ReactElement<any>
    );

    expect(addLayerMock.mock.calls[0]).toEqual([{
      id: 'source-1',
      source: 'source-1',
      type: 'symbol'
    }, undefined]);
  });
...
```

Output

```bash
Source › Should render source with default options

    expect(received).toEqual(expected)

    Expected value to equal:
      [{"id": "source-1", "source": "source-1", "type": "symbol"}, undefined]
    Received:
      undefined

    Difference:

      Comparing two different types of values. Expected array but received undefined.
```

I understand I copied most of the stuff from layer.test.tsx, but I didn't expect it to return always undefined... I don't exactly get how to pass mocked data